January release highlights for JavaScript & TypeScript (#2240)
Azure/azure-sdk: releases/2021-01/js.md

The Azure SDK team is pleased to make available the January 2021 client library release.

#### Updates

- Azure Storage Queue
- Azure Storage Blob
- Azure Storage Data Lake
- Azure Storage File Share
- Azure Service Bus

> 

Reviewer: Didn't these storage packages GA STG 75 features?

Author: @ljian3377 please help 🙏

Reviewer: yes, they did. according to the email I just forwarded: STG75 features GA, new generateSasUrl() API on storage clients and several bug fixes for Data Lake.

@azure/storage-blob@12.4.0
@azure/storage-file-datalake@12.3.0
@azure/storage-file-share @12.4.0
@azure/storage-queue @12.3.0

Author: These libraries have been released as GA before, so I was assuming these are only updates. Betas go on the betas section, so these updates can't be betas, but they can be stable versions. I could be missing something, of course.

#### Updates

- Azure Storage Queue
- Azure Storage Blob

> - Azure File Share

Reviewer 2: ```suggestion
- Azure Storage Data Lake.
- Azure Storage File Share.
```

> - Azure File Datalake

Reviewer 2: ```suggestion
```

##### Major Fixes on @azure/storage-file-datalake@12.3.0

- `DataLakePathClient.move()` now supports source and destination authenticated with SAS. Fixed bug [12758](https://github.com/Azure/azure-sdk-for-js/issues/12758).
- Now you can get the functionality of the root directory via the `DataLakeDirectoryClient` created via `FileSystemClient.getDirectoryClient("")`. Fixed bug [12813](https://github.com/Azure/azure-sdk-for-js/issues/12813).

#### storage-file-share [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/CHANGELOG.md)

This release contains bug fixes to improve quality.

#### storage-queue [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/CHANGELOG.md)

This release contains bug fixes to improve quality.

### Azure Service Bus

#### @azure/service-bus [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/CHANGELOG.md)

We're releasing a patch of the Service Bus client that includes some bug fixes.

##### Major Fixes on @azure/service-bus 7.0.1 and 7.0.2

> - Fix issue where `receiveMessages` might return fewer messages than were received, causing them to be potentially locked or lost.  
>   [#12772](https://github.com/Azure/azure-sdk-for-js/pull/12772) [#12908](https://github.com/Azure/azure-sdk-for-js/pull/12908) [#13073](https://github.com/Azure/azure-sdk-for-js/pull/13073)
> - [Bug Fix] Correlation Rule Filter with the "label" set using the `createRule()` method doesn't filter the messages to the subscription or when a subset of properties are set in the correlation filter.  
>   [#13069](https://github.com/Azure/azure-sdk-for-js/pull/13069)
> - [Bug Fix] Receiving messages from sessions in "receiveAndDelete" mode using the `subscribe()` method stops after receiving 2048 of them and leaves the receiver hanging. The bug has been fixed in [#13178](https://github.com/Azure/azure-sdk-for-js/pull/13178).  
>  Also fixes the same issue that is seen with the `receiveMessages` API when large number of messages are requested or if the method is called in a loop.

Reviewer 3: @HarshaNalluru This looks like a copy of the changelog. What we want to focus on is highlights instead of listing every bug fix and links to PR, because the link to the changelog for more details is already provided

```suggestion
- Improved reliability of the `receiveMessages()` method on the receiver when dealing with large number of messages and slow network connectivity.
- Resolved the issue of hanging receivers when receiving more than 2048 messages from sessions in `receiveAndDelete` mode.
- Fix the use of correlation rule filter when `label` is used
```

Reviewer 4: Recommend changing 
`Fix the use of correlation rule filter when "label" is used`
 to 
`Fix the use of correlation rule filter when a subset of properties are set`.

Author: Fixed here: https://github.com/Azure/azure-sdk/pull/2240/files#diff-513feee854d000bc1c01faa0ff54e534393c4ad89fddbcbf13bb3bdc2d32a684R76 please review

## Release highlights

---

> 

Reviewer 5: ### Azure Tables 

#### @azure/data-tables [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/tables/data-tables/CHANGELOG.md)

##### Breaking Changes on @azure/data-tables@1.0.0-beta.4

- Disabled Edm.DateTime deserialization into a JavaScript Date to avoid losing precision [#12650](https://github.com/Azure/azure-sdk-for-js/pull/12650). Deserialization has been disabled on Edm.DataTime to avoid potential precision loss due to the Table Service supporting higher precision than the Javascript Date object.

##### Major Fixes on @azure/data-tables@1.0.0-beta.4

- Fixed an issue that prevented the use of the SDK against `Azure Storage Emulator` and `Azurite@3.9.0-table-alpha.1`
 [#13165](https://github.com/Azure/azure-sdk-for-js/pull/13165)

Author: I added the data-tables notes here: https://github.com/Azure/azure-sdk/pull/2240/files#diff-513feee854d000bc1c01faa0ff54e534393c4ad89fddbcbf13bb3bdc2d32a684R90 please feel free to send more feedback based on what I added.

## Release highlights

> ---

Reviewer 4: I'm probably releasing 7.0.2 today with a critical bug fix, does it makes sense to say "Major Fixes on @Azure/service-bus 7.0.1 and 7.0.2"?

> ---

Reviewer 4: ### Service Bus

#### `@azure/service-bus` [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/CHANGELOG.md)

##### Major Fixes on `@azure/service-bus` 7.0.1 and 7.0.2

- Fix issue where receiveMessages might return fewer messages than were received, causing them to be potentially locked or lost.
  [#12772](https://github.com/Azure/azure-sdk-for-js/pull/12772) [#12908](https://github.com/Azure/azure-sdk-for-js/pull/12908) [#13073](https://github.com/Azure/azure-sdk-for-js/pull/13073)
- [Bug Fix] Correlation Rule Filter with the "label" set using the `createRule()` method doesn't filter the messages to the subscription or when a subset of properties are set in the correlation filter.
  [#13069](https://github.com/Azure/azure-sdk-for-js/pull/13069)
- [Bug Fix] Receiving messages from sessions in "receiveAndDelete" mode using the `subscribe()` method stops after receiving 2048 of them and leaves the receiver hanging. The bug has been fixed in [#13178](https://github.com/Azure/azure-sdk-for-js/pull/13178). 
 Also fixes the same issue that is seen with the `receiveMessages` API when large number of messages are requested or if the method is called in a loop.

> ---

Author: I added the service-bus notes here: https://github.com/Azure/azure-sdk/pull/2240/files#diff-513feee854d000bc1c01faa0ff54e534393c4ad89fddbcbf13bb3bdc2d32a684R75 please feel free to send more feedback based on what I added.
